foundations/javascript_basics/understanding_errors.md

```diff
@@ -0,0 +1,132 @@
+# Introduction
+
+Reading error messages is an underrated art form. At first glance, many beginners shrink away from error messages as they appear to be “scary” and difficult to understand because they include terms one may not be familiar with. However, error messages provide developers with a treasure trove of knowledge, and tell you everything you need to know about how to resolve them! Being able to parse error messages and warnings without fear will enable you to be able to effectively debug your applications, receive meaningful help from others, and empower yourself to push forward when faced with an error.
+
+# Learning Outcomes
+
+- Name at least three kinds of Javascript errors
+- Identify two parts of an error message that help you find where the error originates
+- Be able to understand how to research and resolve errors
+
+# The Anatomy of an Error
+
+Errors contain crucial information that can assist you in locating the code responsible for the error, determining why you have this error, and resolving the error. Let’s assume we have written the following code:
+
+```js
+const a = “Hello”
+const b = “World”
+
+console.log(c)
+```
+
+This code will run, but it will throw an error. The first part of an error displays the type of error. In this example, we have a `ReferenceError`.
+<img width="615" alt="Screen Shot 2021-05-17 at 12 16 37 PM" src="https://user-images.githubusercontent.com/22967723/118543890-c1dce500-b709-11eb-94f3-bbd6757d4e31.png">
+
+
+We’ll learn more about the different error types later in the lesson. Error types denote the kind of error, which gives us clues about how to address them, and why they may arise.
+
+In this case, a `ReferenceError` is an error that is thrown when one attempts to reference a variable that has not yet been declared. The error message also explains this, by stating that `c is not defined`. Different kinds of errors can have different messages based on what is actually causing the `ReferenceError`. For example, another message you may run into that is also a `ReferenceError` is `ReferenceError: can't access lexical declaration 'X' before initialization`. As we can see, this points to a completely different reason than our original `ReferenceError` above. Understanding both the error type and the error message is crucial to comprehending why you are receiving the error.
+
+The next part of an error gives us the line of code, as well as the file in which you can find the error (in this case, our `script.js`). This allows you to easily navigate to the problematic line in your code. Here, the error originates from the fifth line of `script.js`, which is displayed as a link under the error message with the text `at script.js:5`. If you click this link, most browsers will navigate to the exact line of code and the rest of your script in the Sources tab of the Developer Tools. 
```

Comment:
May want to mention the column as well, that may confuse people as to why there are more than one number.